boilerplate/_lib/common.sh

```diff
@@ -1,3 +1,17 @@
+# When this lib is sourced (which is what it's designed for), $0 is the
+# script that did the sourcing.
+SOURCER=$(realpath $0)
+HERE=${SOURCER%/*}
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+CONVENTION_ROOT=$REPO_ROOT/boilerplate
+
+# Were we sourced from within a convention?
+if [[ "$HERE" == "$CONVENTION_ROOT/"* ]]; then
```

Comment:
Should we have more defensive checks around `$HERE` and `$CONVENTION_ROOT`? Seems like if these are empty we should `exit 1`

Reply:
Can do. (Though we would really have to be trying to make either of those things happen.)